src/diagonal.works/b6/search/tree_test.go

```diff
@@ -163,58 +159,57 @@ func TestTreeListAdvanceOnDeletedIterator(t *testing.T) {
 	//4, 5, 6, 10, 12, 13, 15, 16
 	input := []int{10, 5, 15, 4, 6, 13, 16, 12}
 	cases := []struct {
+		name     string
 		delete   int
 		advance  int
 		ok       bool
 		expected []int
 	}{
-		//		{6, 10, true, []int{10, 12, 13, 15, 16}}, // Happy path
-		//		{6, 6, true, []int{10, 12, 13, 15, 16}},  // Advance to deleted
-		{6, 5, true, []int{10, 12, 13, 15, 16}}, // Advance to previous
-		//		{6, 17, true, []int{}},                   // Advance beyond end
+		{"happy path", 6, 10, true, []int{10, 12, 13, 15, 16}},
+		{"Advance to deleted", 6, 6, true, []int{10, 12, 13, 15, 16}},
+		{"Advance to previous", 6, 5, true, []int{10, 12, 13, 15, 16}},
+		// TODO: Make this test case pass.
+		// {"Advance beyond end", 6, 17, true, []int{}},
```

Comment:
Now you've made these subtests, I'm wondering whether rather than commenting this out, it'd be better to add a `skipped bool` to the struct, and use t.Skip()? Not sure.

Reply:
Mmmm I guess it depends on why we're skipping the tests... in this case I think the body of the test or the parameters need to be fixed, so commenting it out gives it a more TODO-ish feel instead of adding more "official" logic that needs to be removed, maybe? I've left things as they were, but will take a look and see if I can fix the test next.